SatisfactoryModLoader/mod/ModHandler.cpp

```diff
@@ -15,15 +18,110 @@
 namespace SML {
 	namespace Mod {
 
-		void ModHandler::loadMods(const char* startingPath) {
-			this->currentStage = GameStage::CONSTRUCT;
-			// split the path
-			std::string appPath(startingPath);
-			size_t pos = appPath.find_last_of('\\');
-			CreateDirectoryA((appPath.substr(0, pos) + "\\mods").c_str(), NULL); //create the directory if it doesn't exist
-			std::string path = appPath.substr(0, pos) + "\\mods";
+		void ModHandler::extractZips() {
+			Utility::cleanupHiddenDirectories();
+
+			auto path = Utility::getModPath();
+			std::string pathExact = path + "\\";
+
+			for (const auto &entry : std::experimental::filesystem::directory_iterator(path)) {
+				if (std::filesystem::is_directory(entry.path().string())) {
+					continue;
+				}
+
+				if (entry.path().extension().string() == ".zip") {
+					std::string archive = pathExact + entry.path().filename().string();
+					Utility::debug("Processing archive: ", archive);
+
+					ttvfs::Root vfs;
+
+					vfs.AddLoader(new ttvfs::DiskLoader);
+					vfs.AddArchiveLoader(new ttvfs::VFSZipArchiveLoader);
+
+					auto modArchive = vfs.AddArchive(archive.c_str());
+
+					ttvfs::File *vf = modArchive->getFile("data.json");
+					if (!vf || !vf->open("r")) {
+						Utility::error("Failed loading mod data.json in ", archive);
+						continue;
+					}
+
+					std::vector<char> buffer(vf->size());
+					size_t bytes = vf->read(buffer.data(), vf->size());
+					vf->close();
+
+					nlohmann::json dataJson;
+
+					try {
+						dataJson = nlohmann::json::parse(buffer.data());
+					}
+					catch (...) {}
+
+					if (!dataJson.contains("objects")) {
+						Utility::error("data.json missing objects in ", archive);
+						continue;
+					}
+
+					auto objects = dataJson["objects"];
+
+					for (auto it = objects.begin(); it != objects.end(); ++it) {
```

Comment:
Is it possible to infer what the objects are in the zip if an old style data.json exists?

Reply:
Yes, but in general very bad idea